src/drivers/BoostDriver.cpp

```diff
@@ -60,8 +60,8 @@ class CukeBoostLogInterceptor : public ::boost::unit_test::unit_test_log_formatt
     void log_exception_finish( std::ostream& ) {};
 
     void log_entry_start( std::ostream&, log_entry_data const&, log_entry_types /*let*/) {};
-    void log_entry_value( std::ostream&, const_string /*value*/);
-    void log_entry_value( std::ostream&, lazy_ostream const& /*value*/) {};
+    void log_entry_value( std::ostream&, const_string value);
+    void log_entry_value( std::ostream&, lazy_ostream const& value);
```

Comment:
I've just submitted 3ec9d0367b73abfeb0b05eebf612f551e88d9dad which enables warnings for all GCC/Clang builds. I don't think it makes sense to make it dependent on the build type. Adding `-Werror` is something we should do only when the warnings are fixed.